File: 2024/day02/part1.php
<?php
    $filename = "real-input.txt";
    $myfile = fopen($filename, "r"); 

    if($myfile){
        $content = fread($myfile,filesize($filename));
        fclose($myfile);
        $lines = explode("\n",$content);
        $res = 0;

        foreach($lines as $report){
            if(strlen($report) === 0) continue;

            $inc = false;
            $isvalid  = true;
            $levels = explode(" ",$report);
            
            for ($i = 0; $i < count($levels)-1; $i++) {
                $current = $levels[$i];
                $next = $levels[$i+1];

                if($i === 0){
                    $inc = $current < $next;
                }

                $diff;

                if($inc){
                    $diff = $next - $current;
                }else {
                    $diff = $current - $next;
                }

                if($diff < 1 || $diff > 3) {
                    $isvalid = false;
                    break;
                }
            }

            if($isvalid) {
                $res+=1;
                echo "Safe: $report\n";
            }else {
                echo "Unsafe: $report\n";
            }

        };

        echo "Result: $res";
    }else {
        echo "Failed to open the file.";
    }
?>

File: 2024/day03/part1.py
input = open("real-input.txt","r").read()

def solution():
    l,r = 0,0
    res = 0

    while r < len(input):
        current = input[l:r+1]

        if current == "mul":
            start_index = r+1
            open = input[start_index]
            start_index+=1
            n1="" 

            while input[start_index].isdigit(): 
                n1+=input[start_index]
                start_index+=1

            s = input[start_index]
            start_index+=1
            n2= ""

            while input[start_index].isdigit(): 
                n2+=input[start_index]
                start_index+=1

            close = input[start_index]

            if open == "(" and close == ")" and s == ",":
                res+=(int(n1)*int(n2))

            r+=1
            l=r
        elif current in "mul":
            r+=1
        else:
            l+=1
            if l >= r:
                r=l

    return res

print("solution: ",solution())






    
    








